### PR TITLE
Handle the golang source file header based on 'go vet' and fix lint error for GMC

### PR DESCRIPTION
**Pull request by zhlsunshine**
## Description

Change list:
1. handle the golang source file header based on 'go vet'
2. update the header template for golang code generated by `go tool`
3. fix all existing lint error on both manager and router
4. add `patch` action for K8s deployment in yaml file `gmc-manager-rbac.yaml`

## Issues

 `n/a`

## Type of change

List the type of change like below. Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would break existing design and interface)

## Dependencies

n/a

## Tests

Describe the tests that you ran to verify your changes.

### CODE REVIEW

**Review by irisdingbj — APPROVED** [review]

File: microservices-connector/cmd/router/main.go [hunk 1]
```diff
@@ -137,7 +149,12 @@ func getServiceURLByStepTarget(step *gmcv1alpha3.Step, svcNameSpace string) stri
 	return step.ServiceURL
 }
 
-func executeStep(step *gmcv1alpha3.Step, graph gmcv1alpha3.GMConnector, input []byte, headers http.Header) ([]byte, int, error) {
+func executeStep(
+	step *gmcv1alpha3.Step,
```
Comment (mkbhanda) [inline microservices-connector/cmd/router/main.go:153]: Do we want to clean up this "gmc" in our next release to be "mc"?

File: microservices-connector/cmd/router/main.go [hunk 2]
```diff
@@ -146,21 +163,32 @@ func executeStep(step *gmcv1alpha3.Step, graph gmcv1alpha3.GMConnector, input []
...
 	if route.NodeName != "" {
-		stepType = "node"
+		stepType = ServiceNode
 	}
 	log.Info("Starting execution of step", "type", stepType, "stepName", route.StepName)
 	if responseBytes, statusCode, err = executeStep(route, graph, input, headers); err != nil {
 		return nil, 500, err
 	}
 
 	if route.Dependency == gmcv1alpha3.Hard && !isSuccessFul(statusCode) {
```
Comment (mkbhanda) [inline microservices-connector/cmd/router/main.go:184]: another instance of gmc -> mc for future

File: microservices-connector/cmd/router/main.go [hunk 3]
```diff
@@ -91,8 +96,15 @@ func callService(step *gmcv1alpha3.Step, serviceUrl string, input []byte, header
 	defer timeTrack(time.Now(), "step", serviceUrl)
 	log.Info("Entering callService", "url", serviceUrl)
 
+	// log the http header from the original request
+	log.Info("Print the http request headers", "HTTP_Header", headers)
```
Comment (mkbhanda) [inline microservices-connector/cmd/router/main.go:100]: The info message does not need to contain "Print the http request headers", just "HTTP_request_header" enough

**Review by mkbhanda — APPROVED** [review]
Will recreate a little issue to clean up naming (gmc -> mc) for later, to be addressed in Milestone-2"

File: microservices-connector/cmd/router/main.go (hunk 1 above)
Comment (zhlsunshine) [reply microservices-connector/cmd/router/main.go:153]: @mkbhanda sure, will do it.

File: microservices-connector/cmd/router/main.go (hunk 2 above)
Comment (zhlsunshine) [reply microservices-connector/cmd/router/main.go:184]: @mkbhanda sure, got it

File: microservices-connector/cmd/router/main.go (hunk 3 above)
Comment (zhlsunshine) [reply microservices-connector/cmd/router/main.go:100]: Hi @mkbhanda, this operation is just keep the `header` of HTTP request which should be used for handling some feature later.

**Pull request merged**